lib/bone.js

```diff
@@ -172,6 +178,97 @@ class Bone {
+   * @memberof Bone
+   */
+  previousChanged(name) {
+    if (name != null) {
+      if (this.rawUnset.has(name) || this.rawPrevious[name] === undefined) return false;
+      const value = this.attribute(name);
+      const valueWas = this.rawPrevious[name] == null ? null : this.rawPrevious[name];
+      return !util.isDeepStrictEqual(value, valueWas);
+    }
+    const result = [];
+    for (const attrKey of Object.keys(this.constructor.attributes)) {
+      if (this.rawUnset.has(attrKey) || this.rawPrevious[attrKey] === undefined) continue;
+      const value = this.attribute(attrKey); // rawValue == null ? null : rawValue;
+      const valueWas = this.rawPrevious[attrKey] == null ? null : this.rawPrevious[attrKey];
+      if (!util.isDeepStrictEqual(value, valueWas)) result.push(attrKey);
+    }
+    return result.length > 0 ? result : false;
+  }
+
+  /**
+   *
+   * @param {string?} name
+   * @returns {Object: {
+   *  name: [ previousValue, currentValue ]
+   * }} changed attribute(s) that compare(s) to current persisted value(s): {@link Bone.raw} compares to {@link Bone.rawSaved}
+   * @memberof Bone
+   */
+  previousChanges(name) {
+    if (name != null) {
+      if (this.rawUnset.has(name) || this.rawPrevious[name] === undefined) return {};
+      const value = this.attribute(name);
+      const valueWas = this.rawPrevious[name] == null ? null : this.rawPrevious[name];
+      if (util.isDeepStrictEqual(value, valueWas)) return {};
+      return { [name]: [ valueWas, value ] };
+    }
+    const result = {};
+    for (const attrKey of Object.keys(this.constructor.attributes)) {
+      if (this.rawUnset.has(attrKey) || this.rawPrevious[attrKey] === undefined) continue;
+      const value = this.attribute(attrKey);
+      const valueWas = this.rawPrevious[attrKey] == null ? null : this.rawPrevious[attrKey];
+      if (!util.isDeepStrictEqual(value, valueWas)) result[attrKey] = [ valueWas, value ];
+    }
+    return result;
+  }
+
+  /**
+   *
+   *
+   * @param {string?} name
+   * @returns {Object: {
+   *  name: [ savedValue, currentValue ]
+   * }} changed attribute(s) that compare(s) to current persisted value(s): {@link Bone.raw} compares to {@link Bone.rawSaved}
+   * @memberof Bone
+   */
+  changes(name) {
+    if (name != null) {
+      if (this.rawUnset.has(name) || this.rawSaved[name] === undefined) return {};
+      const value = this.attribute(name);
+      const valueWas = this.attributeWas(name);
+      if (util.isDeepStrictEqual(value, valueWas)) return {};
+      return { [name]: [ valueWas, value ] };
+    }
+    const result = {};
+    for (const attrKey of Object.keys(this.constructor.attributes)) {
+      if (this.rawUnset.has(name) || this.rawSaved[attrKey] === undefined) continue;
+      const value = this.attribute(attrKey);
+      const valueWas = this.attributeWas(attrKey);
+      if (!util.isDeepStrictEqual(value, valueWas)) result[attrKey] = [ valueWas, value ];
+    }
+    return result;
+  }
```

Comment:
changed can also be added with 

```suggestion
  }
  
  changed(name) {
    const result = Object.keys(this.changes(name);
    return result.length > 0 ? result : false;
  }
```